File: testwarez/src/androidTest/java/com/pgssoft/testwarez/test/runner/CucumberTestRunner.java
package com.pgssoft.testwarez.test.runner; /**
 * Created by mnicpon on 07/11/2017.
 */

import android.os.Bundle;

import cucumber.api.android.CucumberInstrumentationCore;

public class CucumberTestRunner extends android.support.test.runner.AndroidJUnitRunner {

    private final CucumberInstrumentationCore instrumentationCore = new CucumberInstrumentationCore(this);

    @Override
    public void onCreate(final Bundle bundle) {
        super.onCreate(bundle);
        instrumentationCore.create(bundle);
    }

    @Override
    public void onStart() {
        waitForIdleSync();
        instrumentationCore.start();
    }
}

File: testwarez/src/androidTest/java/com/pgssoft/testwarez/test/cucumber/runner/CucumberTestRunner.kt
package com.pgssoft.testwarez.test.cucumber.runner

/**
 * Created by mnicpon on 07/11/2017.
 */

import android.os.Bundle

import cucumber.api.android.CucumberInstrumentationCore

class CucumberTestRunner : android.support.test.runner.AndroidJUnitRunner() {

    private val instrumentationCore = CucumberInstrumentationCore(this)

    override fun onCreate(bundle: Bundle) {
        super.onCreate(bundle)
        instrumentationCore.create(bundle)
    }

    override fun onStart() {
        waitForIdleSync()
        instrumentationCore.start()
    }
}
